packages/xstate-store/src/store.ts

```diff
@@ -76,20 +61,25 @@ function createStoreCore<
   const transition = logic.transition;
 
   function receive(event: StoreEvent) {
-    let effects: StoreEffect<TEmitted>[];
-    [currentSnapshot, effects] = transition(currentSnapshot, event);
+    const [nextSnapshot, effects] = transition(currentSnapshot, event);
+
+    if (nextSnapshot === currentSnapshot && !effects.length) {
+      return;
+    }
+
+    currentSnapshot = nextSnapshot;
 
     inspectionObservers.get(store)?.forEach((observer) => {
```

Comment:
If I'm reading the XState's code correctly this will now diverge from it in behavior. In XState `@xstate.snapshot` seems to be created for all events - not only the ones that cause a snapshot change.

Reply:
Do you think we should still call the inspector then?

Reply:
Hm, we probably don't have to. It depends on what the `inspect` receiver tries to accomplish and how but it feels OK to omit those calls. I think though this, ideally, should be consistent between XState and the store.

Reply:
Keeping it consistent for now